src/dparse/parser.d

```diff
@@ -1423,9 +1408,6 @@ class Parser
         case tok!"immutable":
             node.first = advance();
             break;
-        default:
```

Comment:
`parseCastQualifier` is always called after a `isCastQualifier` check.  I removed the default case and added an assertion in case someone would one day call this sub parser alone.